spec/rails_helper.rb

```diff
@@ -943,7 +943,11 @@ def global_setting(name, value)
 
   GlobalSetting.stubs(name).returns(value)
 
-  before_next_spec { GlobalSetting.reset_s3_cache! }
+  before_next_spec do
+    SiteSetting.hidden_settings_provider.remove_hidden(name)
+    SiteSetting.shadowed_settings.delete(name)
```

Comment:
Should we also run these 2 lines on line 942 so that if something else leaks in the future, it is cleaned up before as well?